Develop to Master (Task1, Task2, fixes after Task2)

aleksandrbelik opened the pull request.

loginov-rocks commented on .eslintrc:
```diff
@@ -0,0 +1,25 @@
+{
+    "extends": [
+        "airbnb-base",
```
Prefer `airbnb` instead of `airbnb-base`.

aleksandrbelik replied:
@loginov-rocks fixed

loginov-rocks commented on .eslintrc:
```diff
@@ -0,0 +1,25 @@
+    },
+    "rules": {
+      "comma-dangle": ["error", "never"],
+      "class-methods-use-this": ["error", { "exceptMethods": ["render"] }]
```
What is the reason to except render method here?

aleksandrbelik replied:
@loginov-rocks fixed

loginov-rocks commented on package.json:
```diff
@@ -1,15 +1,58 @@
+    "babel-plugin-transform-export-extensions": "^6.22.0",
+    "css-loader": "^2.1.1",
+    "eslint": "^5.15.3",
+    "eslint-config-airbnb-base": "^13.1.0",
```
Prefer `eslint-config-airbnb` instead of `eslint-config-airbnb-base`.

aleksandrbelik replied:
@loginov-rocks fixed

loginov-rocks commented on package.json:
```diff
@@ -1,15 +1,58 @@
+    "webpack-cli": "^3.2.3",
+    "webpack-dev-server": "^3.2.1"
+  },
+  "dependencies": {
```
Webpack & webpack plugins are not runtime dependencies, they should be placed within the "devDependencies" object. Switch appropriate eslint rule off for webpack configurations.

aleksandrbelik replied:
@loginov-rocks removed aslant rule for webpack.
As for "webpack" and "webpack-merge" eslint can't pass test with this rules in "devDepencies". It is generate the follow errors:
2:15  error  'webpack-merge' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
  3:17  error  'webpack' should be listed in the project's dependencies, not devDependencies        import/no-extraneous-dependencies

loginov-rocks replied:
If you want to lint development-related stuff (you should, btw), you can add the following line at the beginning:
```
/* eslint-disable import/no-extraneous-dependencies */
```
OR you can create `configs/.eslintrc` with the following content:

```
{
  "rules": {
    "import/no-extraneous-dependencies": "off"
  }
}
```

In any case development dependencies should never appear in the runtime dependencies list.

aleksandrbelik replied:
@loginov-rocks fixed package json and added appropriate rule to eslint config

loginov-rocks commented:
@aleksandrbelik you just disabled rule for the whole project, please reread my comment once again.

aleksandrbelik commented:
> @aleksandrbelik you just disabled rule for the whole project, please reread my comment once again.

moved eslint rule to webpack configs, added linebreak-style disabling
